src/utilities/Settings.cpp

```diff
@@ -37,25 +37,44 @@ void Settings::setId(int id) { Settings::id = id; }
 bool Settings::isYellow() const { return yellow; }
 
 bool Settings::setYellow(bool yellow) {
-    bool switched = rtt::ai::io::io.switchTeamColorChannel(yellow);
+    bool hasWantedColor = false;
 
-    if (switched) {
-        Settings::yellow = yellow;
-        RTT_INFO(yellow ? "Team color changed: Yellow" : "Team color changed: Blue")
+    if (ai::io::io.obtainTeamColorChannel(yellow)) {
+        // We could obtain the necessary channel
+        this->yellow = yellow;
+        hasWantedColor = true;
     } else {
```

Comment:
Seems like this else statement is not necessary

Reply:
I removed the else statement